### ISSUE TITLE
TestExecution encountered unexpected failures System.IO.FileNotFoundException: Could not load file or assembly 'System.Net.Http, Version=4.2.0.0, or one of its dependencies. The system cannot find the file specified.

### ISSUE THREAD

**Issue by ryanbuening** [microsoft/azure-pipelines-tasks#20513]
### New issue checklist

- [X] I searched for [existing GitHub issues](https://github.com/microsoft/azure-pipelines-tasks/issues)
- [X] I read [pipeline troubleshooting guide](https://docs.microsoft.com/vsts/build-release/actions/troubleshooting)
- [X] I checked [how to collect logs](https://learn.microsoft.com/azure/devops/pipelines/troubleshooting/review-logs?view=azure-devops)

### Task name

Visual Studio Test

### Task version

3.246.1

### Issue Description

I'm using Visual Studio Test with the task version set to 3 and I'm still seeing this error. 

`TestExecution encountered unexpected failures System.IO.FileNotFoundException: Could not load file or assembly 'System.Net.Http, Version=4.2.0.0, or one of its dependencies. The system cannot find the file specified.`

This starting happening after upgrading to .NET 8 and I'm wondering if it has something to do with [this change](https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/8.0/implicit-global-using-netfx).

- I attempted adding `global using System.Net.Http` in my e2e test project, but it still didn't resolve the issue.
- I also attempted adding `System.Net.Http` (4.3.4) as a package reference to my e2e csproj which also didn't resolve the issue.

### Environment type (Please select at least one enviroment where you face this issue)

- [ ] Self-Hosted
- [ ] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Windows

### Relevant log output

```shell
2024-10-04T13:40:13.5788824Z ##[section]Starting: vstest e2e
2024-10-04T13:40:13.5801182Z ==============================================================================
2024-10-04T13:40:13.5801346Z Task         : Visual Studio Test
2024-10-04T13:40:13.5801608Z Description  : Run unit and functional tests (Selenium, Appium, Coded UI test, etc.) using the Visual Studio Test (VsTest) runner. Test frameworks that have a Visual Studio test adapter such as MsTest, xUnit, NUnit, Chutzpah (for JavaScript tests using QUnit, Mocha and Jasmine), etc. can be run. Tests can be distributed on multiple agents using this task (version 2 and later).
2024-10-04T13:40:13.5802001Z Version      : 3.246.1
2024-10-04T13:40:13.5802062Z Author       : Microsoft Corporation
2024-10-04T13:40:13.5802140Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/test/vstest
2024-10-04T13:40:13.5802251Z ==============================================================================
2024-10-04T13:40:14.6345952Z Running tests using vstest.console.exe runner.
2024-10-04T13:40:14.6378548Z ======================================================
2024-10-04T13:40:14.6379531Z Test selector : Test assemblies
2024-10-04T13:40:14.6379912Z Test filter criteria : undefined
2024-10-04T13:40:14.6380383Z Search folder : C:\azagent\A4\_work\r5\a
2024-10-04T13:40:14.6450160Z Action when minimum tests threshold not met : donothing
2024-10-04T13:40:14.6450377Z Minimum tests expected to be run: 0
2024-10-04T13:40:14.7810125Z Run in parallel : false
2024-10-04T13:40:14.7810489Z Run in isolation : true
2024-10-04T13:40:14.7810829Z Path to custom adapters : undefined
2024-10-04T13:40:14.7811734Z Other console options : undefined
2024-10-04T13:40:14.7893148Z Code coverage enabled : false
2024-10-04T13:40:14.7894553Z Diagnostics enabled : true
2024-10-04T13:40:14.7895774Z Rerun failed tests: true
2024-10-04T13:40:14.7896482Z Rerun maximum failed test case limit: 5
2024-10-04T13:40:14.7896949Z Rerun maximum attempts: 3
2024-10-04T13:40:14.7902831Z Run the tests locally using vstest.console.exe
2024-10-04T13:40:14.7903233Z ========================================================
2024-10-04T13:40:14.7904003Z Source filter: **/E2E.Tests.dll
2024-10-04T13:40:14.9017891Z ======================================================
2024-10-04T13:40:14.9392187Z [command]C:\azagent\A4\_work\_tasks\VSTest_ef087383-ee5e-42c7-9a53-ab56c98420f9\3.246.1\Modules\DTAExecutionHost.exe --inputFile C:\azagent\A4\_work\_temp\input_2f9384d0-8256-11ef-b7c5-7f8e9e84dc3d.json
2024-10-04T13:40:15.0325790Z ##########################################################################
2024-10-04T13:40:15.0326175Z DtaExecutionHost version 19.240.34919.1.
2024-10-04T13:40:15.3815098Z Starting TestExecution Model...
2024-10-04T13:40:15.4362691Z ##[error]TestExecution encountered unexpected failures System.IO.FileNotFoundException: Could not load file or assembly 'System.Net.Http, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
2024-10-04T13:40:15.4371501Z File name: 'System.Net.Http, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
2024-10-04T13:40:15.4371821Z    at MS.VS.TestService.Task.Common.VssConnectionHelper.get_Instance()
2024-10-04T13:40:15.4372083Z    at Microsoft.VisualStudio.TestService.AgentExecutionHost.LocalTestRunHost.InitializeVssConnection()
2024-10-04T13:40:15.4372370Z    at Microsoft.VisualStudio.TestService.AgentExecutionHost.LocalTestRunHost.Execute(Boolean debugLogs)
2024-10-04T13:40:15.4372652Z    at Microsoft.VisualStudio.TestService.AgentExecutionHost.Program.Main(String[] args)
2024-10-04T13:40:15.4372775Z 
2024-10-04T13:40:15.4372915Z WRN: Assembly binding logging is turned OFF.
2024-10-04T13:40:15.4373177Z To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
2024-10-04T13:40:15.4373451Z Note: There is some performance penalty associated with assembly bind failure logging.
2024-10-04T13:40:15.4375077Z To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
2024-10-04T13:40:15.4375200Z 
2024-10-04T13:40:15.4384625Z System.IO.FileNotFoundException: Could not load file or assembly 'System.Net.Http, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
2024-10-04T13:40:15.4385264Z    at MS.VS.TestService.Task.Common.VssConnectionHelper.get_Instance()
2024-10-04T13:40:15.4385526Z    at Microsoft.VisualStudio.TestService.AgentExecutionHost.LocalTestRunHost.InitializeVssConnection()
2024-10-04T13:40:15.4385817Z    at Microsoft.VisualStudio.TestService.AgentExecutionHost.LocalTestRunHost.Execute(Boolean debugLogs)
2024-10-04T13:40:15.4385973Z 
2024-10-04T13:40:15.4387026Z ##[warning]Vstest failed with error. Check logs for failures. There might be failed tests.
2024-10-04T13:40:15.4388090Z ##[error]Error: The process 'C:\azagent\A4\_work\_tasks\VSTest_ef087383-ee5e-42c7-9a53-ab56c98420f9\3.246.1\Modules\DTAExecutionHost.exe' failed with exit code 3762504530
2024-10-04T13:40:15.4389962Z ##[error]Vstest failed with error. Check logs for failures. There might be failed tests.
2024-10-04T13:40:15.4569543Z ##[section]Finishing: vstest e2e
```

**Comment by ryanbuening** [comment]
This was labeled with `bug` and `triage`...

**Comment by lesscodetxm** [comment]
We just hit this also, after an upgrade on our ADO build agents. The solution specified in https://github.com/dotnet/docs/issues/46894 didn't seems to apply for us - we don't reference a specific version of System.Net.Http and we have no binding redirects for it. Our vstest unit tests started to fail in our pipelines after the update.
